### PR TITLE
makefile support for arm64 packaging

### PR DESCRIPTION
**Pull request by Zambiorix**
existing targets work as usual

New targets:

```
package_rpm
package_deb

package_amd64
package_rpm_amd64
package_deb_amd64

package_arm64
package_rpm_arm64
package_deb_arm64

package_base_name
```

### CODE REVIEW

**Comment by mperham** [comment]
I don’t think this will build arm64 for release if your goal is released packages.

**Comment by Zambiorix** [comment]
`make package` does not build release packages?

**Comment by mperham** [comment]
It does but where is each arch explicitly built? I don't see where you actually build both arches.

**Comment by Zambiorix** [comment]
`make package` now builds packages for both amd64 and arm64
I have split the `clean` target, to perform some cleaning between architecture builds and making sure the output folder is not deleted.

File: Makefile [hunk 1]
```diff
@@ -104,7 +106,12 @@ cover:
 xbuild: clean generate
 	@CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o $(NAME) cmd/faktory/daemon.go
 	# brew install upx
-	upx -qq ./faktory
+	upx -qq ./$(NAME)
+
+xbuild_arm64: clean generate
```
Comment (mperham) [inline Makefile:111]: This should not be a separate target. xbuild should build both.

File: Makefile [hunk 2]
```diff
@@ -133,14 +140,20 @@ fmt: ## Format the code
 work: ## Run a simple Ruby worker, see also "make run"
 	cd test/ruby && bundle exec faktory-worker -v -r ./app.rb -q critical -q default -q bulk
 
-clean: ## Clean the project, set it up for a new build
+clean_project: ## Clean the project, set it up for a new build
```
Comment (mperham) [inline Makefile:143]: Needs a better name, clean and clean_project is just confusing.

File: Makefile [hunk 3]
```diff
@@ -183,6 +200,22 @@ rpm: xbuild
 		faktory=/usr/bin/faktory \
 		packaging/root/=/
 
+rpm_arm64: xbuild_arm64
```
Comment (mperham) [inline Makefile:203]: This should be part of rpm, not a separate target.

File: Makefile [hunk 4]
```diff
@@ -199,6 +232,22 @@ deb: xbuild
 		faktory=/usr/bin/faktory \
 		packaging/root/=/
 
+deb_arm64: xbuild_arm64
```
Comment (mperham) [inline Makefile:235]: Ditto.

File: Makefile (hunk 1 above)
Comment (Zambiorix) [reply Makefile:111]: but they have the same name...
I can put them in a separte folder, perhaps a subfolder in the output folder?
Is cleaning between arch builds required?

Comment (mperham) [reply Makefile:111]: Arch subfolders seems fine.

**Comment by Zambiorix** [comment]
Ok, xbuild now builds both archs as `faktory_amd64` and `faktory_arm64`
`deb` and `rpm` force copy the output to `faktory` to keep the package names intact.
The both first start with arm64 and then amd64, leaving the last one undeleted, mimicking  the same behavior of this makefile before the change.

`clean` is updated

`.gitignore` is updated

**Comment by mperham** [comment]
Thank you, nice work!

**Pull request merged**